src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -77,50 +76,21 @@ public class TopicPartitionChannel {
 
   private final Lock bufferLock = new ReentrantLock(true);
 
-  /**
-   * States whether this channel has received any records before.
-   *
-   * <p>If this is false, the topicPartitionChannel has recently been initialised and didnt receive
-   * any records before or TopicPartitionChannel was recently created. (Or may be there was some
-   * error in some API and hence we had to fetch it again)
-   *
-   * <p>If the channel is closed, or if partition reassignment is triggered (Rebalancing), we wipe
-   * off partitionsToChannel cache in {@link SnowflakeSinkServiceV2}
-   */
-  private boolean hasChannelReceivedAnyRecordsBefore = false;
-
   // This offset is updated when Snowflake has received offset from insertRows API
   // We will update this value after calling offsetToken API for this channel
-  // We will only update it during start of the channel initialization
-  private AtomicLong offsetPersistedInSnowflake =
+  private final AtomicLong offsetPersistedInSnowflake =
+      new AtomicLong(NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE);
+
+  // This offset is updated when KC processed an offset, it's used to make sure that we won't
+  // process records that are already processed. It will be reset to the committed offset every time
+  // we fetch a valid offset token from Snowflake
```

Comment:
> It will be reset to the committed offset every time we fetch a valid offset token from Snowflake

Can you explain this further? here `committedOffset` is terminology picked from our getLatestComm... API?

Reply:
Yeah committed offset is Snowflake terminology.